app/src/main/kotlin/edu/stanford/bdh/engagehf/bluetooth/BluetoothViewModel.kt

```diff
@@ -218,9 +223,20 @@ class BluetoothViewModel @Inject internal constructor(
         messageId: String,
         isDismissible: Boolean,
     ) {
+        var shouldDismissMessage = isDismissible
         when (messagesAction) {
             is MessagesAction.HealthSummaryAction -> {
-                handleHealthSummaryAction(messageId)
+                handleHealthSummaryAction(messageId).onFailure {
+                    shouldDismissMessage = false
+                    return
```

Comment:
return is also not needed anymore imo, either return only + comment as it was before or address via the new `shouldDismissMessage` variable which will skip dismissing below

Reply:
i left the `return` in because i thought it would make it easier to read, that nothing more would happen in this case, but i removed it now